src/app.py

```diff
@@ -511,9 +509,18 @@ def last_activity_to_datetime(use_example_data: bool, x: str | None) -> str | No
     # Convert date column from str to datetime
     df_historic_data["day"] = df_historic_data["day"].apply(lambda x: datetime.strptime(x, "%Y-%m-%d"))
 
+    # Create a date picker for the user to select the date range
+    starting_date = st.date_input("Enter starting date", date.today() - timedelta(30))
```

Comment:
It's probably best to have the starting date as the minimum date within the dataset. When the user first opens the dashboard, we want them to see all the data available.